### ISSUE TITLE
high resolution marker not working on macOS?

### ISSUE THREAD

**Issue by henryiii** [piccolomo/plotext#32]
My plots are all showing up with `s`'s on macOS rather than the small character, which I'm sure the terminal supprorts. I'm not sure why, looking into the code, as the platform should be set to "linux" as far as I can tell. However, I noticed a few things that made it hard to parse and debug the code:

* The platform should be called "unix", not "linux", if it encompasses both (it should)
* There should not be a function called "plot" and a module called "plot" at the same level. It's painfully hard to debug and access the module called "plot" because of the function. Could you rename the module to `_plot`, for example?
* `import *` should be avoided in a library. You only have one, so that's not too bad; but two or more make it really hard to track things down.
* The path manipulation in __init__.py is unnessisary. 
* _Never_, never tell anyone to `sudo pip` (in the readme). Recent versions of pip will install into the user location if the system location is not writable (but, hopefully, all users will be in virtual environments). Pip literally prints a warning if it can detect you are running it with sudo, because it _will_ break your system. Could you touch up the README suggestion?
* There are no git tags. I can't tell what's in the latest release, and what's been added since then.
* Couldn't you tests check the output? That would be really helpful in GHA, where you can run the tests with 
* By avoiding NumPy, the histogram function (for example) is painfully slower than it needs to be. Couldn't that be a minimal requirement? If not, then maybe try to import it, and only use a faster implementation if it's available?
* The histogram function could default to making a number of bins equal to the available width, which would be really nice, I think (or 2x the available width in small mode).

Hopefully these points are helpful. :) You might find the guide here https://scikit-hep.org/developer helpful for PyTest and nice packaging / checks. Anyway, the main problem is my "small" marker plots have only s's.

**Comment by piccolomo** [comment]
Hi @henryiii,

thanks for you message. Le me answer each point:

- I will follow this recommendation for next version, thanks. I have no way to test the package on a MacOS. I will update when next version is ready: perhaps you could help me test it?
- I may call the `plot.py` file something like `definitions.py` or `core.py`
- Any recommendation on what to use instead of `import *`?
- How would you change the path manipulation in the file `__init__.py`?
- about `sudo` before `pip`: got it!
- I don't know what git tags are. Could you tell me how to add them?
- I don't know what GHA is, but I have created a `plt.test()` function useful to see if the package works in any particular system.
- I may be thinking on using `numpy` for next version. I wanted to keep the importing to a minimum but I also noticed that histogram and bar plots could get slow.
- I don't understand this question as I don't know what you refer with "available width"

Thanks for feedback

**Comment by henryiii** [comment]
Hi @piccolomo,

* I am happy to test, let me know when. :) I might be able to help off and on, too, let me know if there's something you want me to try setting up.
* `core.py` sounds good to me, but `definitions` is better than `plot` when you have a `plot` function. :)
* I'd import explicitly. Then someone reading your file knows exactly where each thing comes from. It's a little more trouble to keep up, but this is a library, not the REPL. So `from plotext.core import (item_1, item_2, ...)`.
* You don't need it. You can always import from a file in the same directory.
* :)
* The easiest way is to make a "GitHub release" from the web. Tell it what branch you are using, and it makes a tag for you - name it `vX.Y.Z`. If you want to make older commits into releases, or you prefer manually making tags, `git tag -a vX.Y.Z -m "Version blablabla" && git push --tags` will do it.
* GitHub Actions - it's a system that runs on every commit and does things for you. You can read about it in the page I linked (or lots of other places). If you have tests that check the output, then you can have a macOS, linux, and Windows runner run the tests and see if it is really producing the output you expect. I could easily set it up, though the `plt.test()` function doesn't do very much that's likely to not work, as it's mostly just a visual printout, IIRC.
* NumPy is pretty much always available. Pythonista (iOS) has it, macOS ships with it, Blender ships with it, can't really think of a place where you can't get NumPy. If you add it to install_requires, I think that's completely safe.
* If you are computing a histogram, and you have an active area 23 blocks wide, the perfect number of bins would be 46 (small) or 23 (large). This might be a little tricky because it could depend on how wide the bin count numbers on the far left are.

**Comment by piccolomo** [comment]
Hi @henryiii ,

the new 4.0.0 version of `plotext` is now available on [github](https://github.com/piccolomo/plotext) and [pypi](https://pypi.org/project/plotext/). :-)

I have tested it on Linux and Windows, but not on Mac OS. It would be kind of you, if you could run a test and give a report. 

I have also implemented your other recommendations, including the one regarding `unittest`: the test is called `test,py` and is in the main `plotext` code directory. 

I have not used git tags, as honestly I am now really sure what they are. 
I have not used `numpy`: I have done some quick test and didn't see much difference in terms of computational time in using `numpy`. The main function that needs an update is `plotext/_matrices.py/subplot_matrices/update()`, not sure if it will make a difference, but feel free to give your opinion.

Regarding the histogram size idea: I cannot tell in general a plotting function its size, as it can only be specified with the `plt.plotsize()` function and then later actually calculated with `plt.show()`. In general cases the plot size set could be different then the final one (due to terminal restriction) and so it is evaluated during the building of the actual plot. 

Thanks and all the best,
Savino

**Comment by piccolomo** [comment]
Hi @henryiii,

I close the issue for lack or reply, but feel free to reopen it if necessary. 

Thanks for help and all the best,
Savino

P.S. newer 4.1.0 version of `plotext` is now available on [github](https://github.com/piccolomo/plotext) and [pypi](https://pypi.org/project/plotext/). :-)
